src/common.d.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, PureComponent, FC } from "react"
+
+export declare type SubscribeFn<T> = (data: T) => void
+
+export declare type UnsubscribeFn = () => void
+
+export declare type MapFn<T> = (data: T) => T | any
+
+export declare type GenericReactComponent<T, S, SS> = (
```

Comment:
Probably `Generic` in the type name is redundant.
I don't see any non-Generic component type :)